server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -876,61 +853,53 @@ private void cancelOfflineTableMigrations(KeyExtent extent) {
+      if (level == Ample.DataLevel.ROOT) {
+        RootTabletMetadata rtm = RootTabletMetadata.read(manager.getContext());
+        entries = rtm.getKeyValues();
+      } else {
+        Scanner scanner =
+            manager.getContext().createScanner(level.metaTable(), Authorizations.EMPTY);
+        scanner.fetchColumnFamily(CurrentLocationColumnFamily.NAME);
+        scanner.fetchColumnFamily(FutureLocationColumnFamily.NAME);
+        scanner.setRange(new Range(extent.toMetaRow()));
+        entries = scanner.stream();
       }
-      Scanner scanner = manager.getContext().createScanner(table, Authorizations.EMPTY);
-      scanner.fetchColumnFamily(CurrentLocationColumnFamily.NAME);
-      scanner.fetchColumnFamily(FutureLocationColumnFamily.NAME);
-      scanner.setRange(new Range(row));
-      for (Entry<Key,Value> entry : scanner) {
+
+      entries.forEach(entry -> {
         if (entry.getKey().getColumnFamily().equals(CurrentLocationColumnFamily.NAME)) {
           assigned.put(entry.getKey(), entry.getValue());
         } else if (entry.getKey().getColumnFamily().equals(FutureLocationColumnFamily.NAME)) {
           future.put(entry.getKey(), entry.getValue());
         }
-      }
-      if (!future.isEmpty() && !assigned.isEmpty()) {
-        Manager.log.warn("Found a tablet assigned and hosted, attempting to repair");
-      } else if (future.size() > 1 && assigned.isEmpty()) {
-        Manager.log.warn("Found a tablet assigned to multiple servers, attempting to repair");
-      } else if (future.isEmpty() && assigned.size() > 1) {
-        Manager.log.warn("Found a tablet hosted on multiple servers, attempting to repair");
+      });
+
+      var count = future.size() + assigned.size();
+      if (count <= 1) {
+        Manager.log.info("Tablet {} seems to have correct location based on inspection",
+            tabletMetadata.getExtent());
       } else {
-        Manager.log.info("Attempted a repair, but nothing seems to be obviously wrong. {} {}",
-            assigned, future);
-        return;
-      }
-      Iterator<Entry<Key,Value>> iter =
-          Iterators.concat(future.entrySet().iterator(), assigned.entrySet().iterator());
-      while (iter.hasNext()) {
-        Entry<Key,Value> entry = iter.next();
-        TServerInstance alive = manager.tserverSet.find(entry.getValue().toString());
-        if (alive == null) {
-          Manager.log.info("Removing entry  {}", entry);
-          BatchWriter bw = manager.getContext().createBatchWriter(table);
-          Mutation m = new Mutation(entry.getKey().getRow());
-          m.putDelete(entry.getKey().getColumnFamily(), entry.getKey().getColumnQualifier());
-          bw.addMutation(m);
-          bw.close();
-          return;
+        for (Map.Entry<Key,Value> entry : future.entrySet()) {
+          TServerInstance alive = manager.tserverSet.find(entry.getValue().toString());
+          Manager.log.warn("Saw duplicate future location key:{} value:{} alive:{} ",
```

Comment:
I wonder if these warns should be debug.

Reply:
The fact of the problem is logged at error.  These are just logging details about the problem so dropping them to debug would be good.